components/affix/index.tsx

```diff
@@ -60,117 +59,47 @@ export interface AffixState {
   prevTarget: Window | HTMLElement | null;
 }
 
-class InternalAffix extends React.Component<InternalAffixProps, AffixState> {
-  static contextType = ConfigContext;
-
-  state: AffixState = {
-    status: AffixStatus.None,
-    lastAffix: false,
-    prevTarget: null,
-  };
-
-  private placeholderNodeRef = createRef<HTMLDivElement>();
-
-  private fixedNodeRef = createRef<HTMLDivElement>();
-
-  private timer: ReturnType<typeof setTimeout> | null;
-
-  context: ConfigConsumerProps;
-
-  private getTargetFunc() {
-    const { getTargetContainer } = this.context;
-    const { target } = this.props;
-
-    if (target !== undefined) {
-      return target;
-    }
-
-    return getTargetContainer ?? getDefaultTarget;
-  }
-
-  addListeners = () => {
-    const targetFunc = this.getTargetFunc();
-    const target = targetFunc?.();
-    const { prevTarget } = this.state;
-    if (prevTarget !== target) {
-      TRIGGER_EVENTS.forEach((eventName) => {
-        prevTarget?.removeEventListener(eventName, this.lazyUpdatePosition);
-        target?.addEventListener(eventName, this.lazyUpdatePosition);
-      });
-      this.updatePosition();
-      this.setState({ prevTarget: target });
-    }
-  };
-
-  removeListeners = () => {
-    if (this.timer) {
-      clearTimeout(this.timer);
-      this.timer = null;
-    }
-    const { prevTarget } = this.state;
-    const targetFunc = this.getTargetFunc();
-    const newTarget = targetFunc?.();
-    TRIGGER_EVENTS.forEach((eventName) => {
-      newTarget?.removeEventListener(eventName, this.lazyUpdatePosition);
-      prevTarget?.removeEventListener(eventName, this.lazyUpdatePosition);
-    });
-    this.updatePosition.cancel();
-    // https://github.com/ant-design/ant-design/issues/22683
-    this.lazyUpdatePosition.cancel();
-  };
+interface AffixRef {
+  updatePosition: ReturnType<typeof throttleByAnimationFrame>;
+}
 
-  // Event handler
-  componentDidMount() {
-    // [Legacy] Wait for parent component ref has its value.
-    // We should use target as directly element instead of function which makes element check hard.
-    this.timer = setTimeout(this.addListeners);
-  }
+const InternalAffix = React.forwardRef<AffixRef, InternalAffixProps>((props, ref) => {
+  const [affixStyle, setAffixStyle] = React.useState<React.CSSProperties>();
+  const [placeholderStyle, setPlaceholderStyle] = React.useState<React.CSSProperties>();
 
-  componentDidUpdate(prevProps: AffixProps) {
-    this.addListeners();
-    if (
-      prevProps.offsetTop !== this.props.offsetTop ||
-      prevProps.offsetBottom !== this.props.offsetBottom
-    ) {
-      this.updatePosition();
-    }
-    this.measure();
-  }
+  const status = React.useRef(AffixStatus.None);
```

Comment:
This should all be controlled by `useState` instead of `useRef`

Reply:
`status` and `lastAffix` are intermediate variables used in calculating the style, rather than being directly consumed by render. I think using `useRef` could help avoid unnecessary re-renders. 

What do you think? I'm open to discussing this further. 
Please let me know if you would like me to clarify or expand on anything.

Reply:
@zombieJ ping